### DISCUSSION TITLE
Error 500/Error [BSONError]

### DISCUSSION THREAD

**Discussion by HenrikeStahlhut** [neuefische/questions#612] [Web] [answered]
Hi everyone ✌🏻,

**Short introduction:**
I want to have an overview of all plants in the database. To have that, i `mapped` over the fetched `data` just as i did when i was working on displaying the rooms as a list. I do this with the following code: 

```
export default function AllPlantsList() {
  const { data: plants, error, isLoading } = useSWR("/api/plants", fetcher);

  if (error) {
    return (
      <StyledError>
        <StyledErrorH3>ERROR</StyledErrorH3>Failed to load plants
      </StyledError>
    );
  }

  if (isLoading) {
    return <StyledLoading>Loading your plants ...</StyledLoading>;
  }

  return (
    <>
      <StyledList>
        {plants.map((plant) => (
          <StyledCard key={plant._id}>
            <StyledListItem key={plant._id}>
              🪴 {plant.title} - Room: {plant.room}
            </StyledListItem>
          </StyledCard>
        ))}
      </StyledList>
    </>
  );
}

```

**The plant database looks like this:** 
<img width="360" alt="Screenshot 2023-05-23 at 12 52 04" src="https://github.com/neuefische/questions/assets/126799897/e6a7ff7c-e7ff-4d07-b25c-a396fda8e3be">

**Problem:** 
The plants are displayed just as i wanted them to, but i get the following errors: 

![Screenshot 2023-05-23 at 12 53 29](https://github.com/neuefische/questions/assets/126799897/1d97d1e9-b820-431a-a3cc-0f0400ed5f23)

![Screenshot 2023-05-23 at 12 54 51](https://github.com/neuefische/questions/assets/126799897/73d7c85a-8943-454b-b2bf-01c271a82653)

Since the error references the fetcher, heres the code: 

```
export const fetcher = async (...args) => {
  const response = await fetch(...args);
  if (!response.ok) {
    throw new Error(`Request with ${JSON.stringify(args)} failed.`);
  }
  return await response.json();
};
```

I don't really know what i did wrong, because the code is basically identical to other times when fetching/mapping/etc. worked. 

- [Vercel](https://capstone-project-git-feat-plants-in-room-henrikestahlhut.vercel.app/)
- [Repo](https://github.com/HenrikeStahlhut/capstone-project)

I would appreciate any ideas or solutions 🌞

**Comment by onemanwenttomow** [comment]
@HenrikeStahlhut I can't seem to find the code you reference above in your [repo](https://github.com/HenrikeStahlhut/capstone-project). 

Is it on a branch that hasn't been pushed to github yet?

**Comment by HenrikeStahlhut** [comment]
@onemanwenttomow Ah yes, i forgot to push the branch. The code is [here](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/components/AllPlantsList/AllPlantsList.js)

**Reply by onemanwenttomow** [reply, marked as answer]
No worries. 

I don't think the problem is in `AllPlantsList`.

I think we need to move "up" and look at the following. 

[`/pages/plants/all/index.js`](https://github.com/HenrikeStahlhut/capstone-project/blob/feat/all-plants-overview/pages/plants/all/index.js#L12)

I think the problem is the `PlantList` component. It is also making a request with `useSWR`, but it is expecting a room prop to be passed to that it can find plants by room (i guess). At the moment it is not being passed that prop (I am not sure if you want it here at all), but that means it is making the following request. 

`/api/plants/undefined`

Where the `undefined` is what is causing the problem. 

If you remove it, the error should go away. 

``` js
import AllPlantsList from "@/components/AllPlantsList/AllPlantsList";
import BackButton from "@/components/BackButton/BackButton";
import Header from "@/components/Header/Header";
import PlantList from "@/components/PlantList/PlantList";

export default function AllPlants() {
  return (
    <>
      <BackButton href="/" />
      <Header>Your Plants</Header>
      <AllPlantsList />
      {/* <PlantList /> */}
    </>
  );
}
```

**Comment by HenrikeStahlhut** [comment]
@onemanwenttomow THANKS! That was the problem, i don't know why i added PlantList to the index. Guess i mixed something up.
